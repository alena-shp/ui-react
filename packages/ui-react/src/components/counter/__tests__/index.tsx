import React from 'react';
import { create } from 'react-test-renderer';

import { Counter } from '..';

test('Default', () => {
    const tree = create(
        <Counter children="99" />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

test('Interactive', () => {
    const tree = create(
        <Counter
            children="11"
            onClick={jest.fn()}
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

test('Gray', () => {
    const tree = create(
        <Counter
            children="99"
            presetStyle="dark-gray"
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();

    tree.update(
        <Counter
            children="99"
            presetStyle="dark-gray"
            presetSize="big"
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

test('Orange', () => {
    const tree = create(
        <Counter
            children="99"
            presetStyle="orange"
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();

    tree.update(
        <Counter
            children="99"
            presetStyle="orange"
            presetSize="big"
        />
    );

    expect(tree.toJSON()).toMatchSnapshot();
});
