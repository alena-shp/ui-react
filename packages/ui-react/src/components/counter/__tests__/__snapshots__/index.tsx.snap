// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default 1`] = `
<span
  className="counter counter_preset-style-orange counter_preset-size-small"
>
  99
</span>
`;

exports[`Gray 1`] = `
<span
  className="counter counter_preset-style-dark-gray counter_preset-size-small"
>
  99
</span>
`;

exports[`Gray 2`] = `
<span
  className="counter counter_preset-style-dark-gray counter_preset-size-big"
>
  99
</span>
`;

exports[`Interactive 1`] = `
<span
  className="counter counter_preset-style-orange counter_preset-size-small counter_interactive"
  onClick={[MockFunction]}
>
  11
</span>
`;

exports[`Orange 1`] = `
<span
  className="counter counter_preset-style-orange counter_preset-size-small"
>
  99
</span>
`;

exports[`Orange 2`] = `
<span
  className="counter counter_preset-style-orange counter_preset-size-big"
>
  99
</span>
`;
